### PR TITLE
Fix kfp_ray_components publish.

### LINKED ISSUE #98: [Bug] make publish in kfp_ray_components is not logging into quay.io
### Search before asking

- [X] I searched the [issues](https://github.com/IBM/data-prep-lab/issues) and found no similar issues.

### Component

KFP workflows

### What happened + What you expected to happen

When using the make file to publish kfp-data-processing image, it fails because I'm not authorized.  Yet I have the DPK_DOCKER_REGISTRY/USER env vars as are supported by transform publishing.  This should be logging the user in with the env vars.

### Reproduction script

```
$ cd kfp/kfp_ray_components
$ make image
$ make publish
docker push quay.io/dataprep1/data-prep-kit/kfp-data-processing:0.1.0
Getting image source signatures
Copying blob sha256:8b2a55f07696d6b9f3367d7d77577a5d8c589d358ca6c89cb8ddf9ea9bdb80d4
Copying blob sha256:6d6a68f624bbd6e78237ef8ee1d6e4ae37c105c17bdc1c9d0c61028592e4cce0
Copying blob sha256:28da0445c4497f3ecb56288bd74d91ed1ff6f86578d1d0f6f9cb2781915163b1
Copying blob sha256:5f70bf18a086007016e948b04aed3b82103a36bea41755b6cddfaf10ace3c6ef
Copying blob sha256:696418cefb060fa658e29b0fe7cf074a7581b558e859d2360cbb6cb90935fbad
Copying blob sha256:0f8b5c9f2bc803c9ef57537e42bf1f977855314e8d463080ab5a20b80f9cc910
Copying blob sha256:9d43689335e2980880d13a36c4aeb151080ff818c52de116f30128470fbe7f96
Copying blob sha256:067112d5bafe7f67682dd678380f426761fe83d1fe17f23ad89d8cb7d34d9ae1
Copying blob sha256:ce61675dbc5e9669b25c4d5b5d753287910ce550b50cec16fd7322e905990e3a
Copying blob sha256:5cc6f1238fac5ae20d628a3b0928c1251c18ad58ca1c3405d1ebf64060a3ffe1
Copying blob sha256:0166a942e7ce9422b37d51ffddef5f0254adc395504b7b4d87fa97615eb75ffe
Copying blob sha256:f900ef5b975c529938bb9c8da081f25c5e33370dce2103883589e8c29cb500ec
Copying blob sha256:e2d425d70ee0fd0d997ab5b84c36cc36927238cc2bc0e5cdcc4195d9b769c25e
Error: writing blob: initiating layer upload to /v2/dataprep1/data-prep-kit/kfp-data-processing/blobs/uploads/ in quay.io: unauthorized: access to the requested resource is not authorized
make: *** [publish] Error 125

```

### Anything else

_No response_

### OS

MacOS (limited support)

### Python

3.10.x

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by revit13**
/Closes #98 

This PR fixes publishing kfp component image to login to quay.io.
It also adds reconcile-requirements rule to kfp transform_workflows.

### CODE REVIEW

File: kfp/kfp_ray_components/Dockerfile [hunk 1]
```diff
@@ -2,6 +2,8 @@ FROM docker.io/rayproject/ray:2.9.3-py310
 
 ARG BUILD_DATE
 ARG GIT_COMMIT
+ARG DPK_LIB_VERSION
+ARG DPK_LIB_KFP_VERSION
```
Comment (blublinsky) [inline kfp/kfp_ray_components/Dockerfile:7]: Why do we need this?

File: kfp/kfp_ray_components/Makefile [hunk 2]
```diff
@@ -2,23 +2,26 @@
...
 
 # Create the docker image making sure the preloaded models are available to copy into the image
 .kfp_comp.image:: Dockerfile requirements.txt
 	@# Help: Build the docker image using the $(DOCKER_FILE) and requirements.txt 
 	$(call check_defined, DOCKER_HOSTNAME)
 	sed -i.back "s/data-prep-toolkit-kfp==[0-9].*/data-prep-toolkit-kfp==${DPK_LIB_KFP_VERSION}/" requirements.txt
-	$(DOCKER) build -t ${IMG} \
+	$(DOCKER) build -t ${DOCKER_IMG} \
+		--build-arg DPK_LIB_VERSION=${DPK_LIB_VERSION} \
+		--build-arg DPK_LIB_KFP_VERSION=${DPK_LIB_KFP_VERSION} \
```
Comment (blublinsky) [inline kfp/kfp_ray_components/Makefile:24]: again, why do we need library versions here?

File: kfp/kfp_ray_components/Dockerfile (hunk 1 above)
Comment (revit13) [reply kfp/kfp_ray_components/Dockerfile:7]: It appears as labels in the image so when doing:
```
docker inspect quay.io/dataprep1/data-prep-kit/kfp-data-processing:0.1.0
```

we get:
```
            "Labels": {
                "build-date": "2024-05-09T16:28:53Z",
                "data-prep-toolkit": "0.0.1",
                "data-prep-toolkit-kfp": "0.0.1",
                "git-commit": "1346e4c",
                "org.opencontainers.image.ref.name": "ubuntu",
                "org.opencontainers.image.version": "20.04"
            },
````

**Review by daw3rd — APPROVED** [review]
THis is ok, but eventually I think we should be building these images using repo source and not pypi

**Pull request merged**